File: Easy/FindTheIndexOfTheFirstOccurrenceInAString.py
class Solution(object):
    def strStr(self, haystack, needle):
        """
        :type haystack: str
        :type needle: str
        :rtype: int
        """
        for i in range(0, len(haystack) - len(needle) + 1):
                if needle[0] == haystack[i]:
                    if all(needle[j - i] == haystack[j] for j in range(i, len(needle) + i)):
                         return i
        return -1
                           
    

haystack = "hello"
needle = "ll"

print(Solution.strStr(Solution, haystack, needle))

File: Easy/FindTheIndexOfTheFirstOccurrenceInAString.java
package Easy;

public class FindTheIndexOfTheFirstOccurrenceInAString {

    public int strStr(String haystack, String needle) {
       
        int l =0;
        int r =0;
        char[] hayArray = haystack.toCharArray();
        char[] needleArray = needle.toCharArray();
        if (needleArray.length > hayArray.length) {
            return -1;
        }
        while(l <= hayArray.length - 1) {
            r = l;
            for (int i = 0; i < needleArray.length && r < hayArray.length; i++) {
                if (hayArray[r] != needleArray[i]) {
                    break;
                } 
                if (i == needleArray.length - 1) {
                    return l;
                }
                r++;
            }
            l++;  
        }
        return -1;
    }

    public static void main(String[] args) {
        FindTheIndexOfTheFirstOccurrenceInAString s = new FindTheIndexOfTheFirstOccurrenceInAString();
        String haystack = "mississippi";
        String needle = "issipi";
        System.out.println(s.strStr(haystack, needle));
    }
}
